experimental/grpc_attestation/src/attestation.rs

```diff
@@ -101,18 +105,21 @@ impl RequestHandler {
 }
 
 /// gRPC Attestation Service implementation.
-pub struct AttestationServer {
+pub struct AttestationServer<F> {
     /// PEM encoded X.509 certificate that signs TEE firmware key.
     tee_certificate: Vec<u8>,
     /// Processes data from client requests and creates responses.
-    request_handler: fn(&[u8]) -> Vec<u8>,
+    request_handler: F,
 }
 
-impl AttestationServer {
-    pub fn create(
-        tee_certificate: Vec<u8>,
-        request_handler: fn(&[u8]) -> Vec<u8>,
-    ) -> anyhow::Result<Self> {
+impl<F, S> AttestationServer<F>
+where
+    F: Send + Sync + Clone + FnOnce(Vec<u8>) -> S,
+    S: std::future::Future<Output = anyhow::Result<Vec<u8>>>
+        + std::marker::Send
+        + std::marker::Sync,
```

Comment:
Previously in `impl<F, S> RequestHandler<F>` you didn't need `std::marker::`

Reply:
Thanks. Fixed.